### ISSUE TITLE
forge-sensei code-generation curriculum

### ISSUE THREAD

**Issue by 4bidden** [ncmlabs/forge#166]
## Context
Extend forge-sensei knowledge base with code-generation-specific training data so toolkit agents can inherit structured generation patterns. forge-sensei + forge-wiki = the school.

## Acceptance Criteria
- [ ] Pre-train sensei on categorized code patterns:
  - `TASKS`: all task examples from conformance + wiki + sensei
  - `FLOWS`: all flow examples (dev-cycle.forge, wiki flows)
  - `AGENTS`: all agent examples (sensei, wiki agents, sentinel)
  - `SYSTEMS`: system wiring patterns
  - `ERRORS`: common compiler error → fix mappings
  - `TESTING`: conformance test structure and patterns
- [ ] Each entry categorized for filtered knowledge transfer
- [ ] `scripts/pretrain-toolkit.sh` — bulk ingest script
- [ ] Verify: `recall "FORGE task declaration"` returns relevant examples with confidence > 0.6
- [ ] Document knowledge categories in forge-sensei README

## Part of
Phase 2 — Milestone 3: Knowledge School

**Comment by 4bidden** [comment]
## Implementation Progress

### Deliverables Created

1. **`docs/forge-sensei.md`** — Comprehensive forge-sensei documentation (19 sections)
   - Architecture, types, events, states, pure functions, tasks, flows, contract
   - The sensei agent, specialist agent, warden
   - Knowledge system (store config, learning modes, TF-IDF recall, 12 categories)
   - Pre-training pipeline (both scripts), continuous learning architecture
   - Claude Code integration, web interface, testing (3 layers), command reference
   - How toolkit agents consume knowledge via categorized recall

2. **`scripts/pretrain-toolkit.sh`** — Categorized code-pattern curriculum (~65 facts)

### Curriculum Summary

| Phase | Category | Facts | What's Covered | Consuming Agent |
|-------|----------|-------|---------------|-----------------|
| 1 | TASKS | 16 | Basic task, reason, classify, exec, compose, skill, pure (4 patterns), composition, use, fn main, templates, search | TaskGenerator #169 |
| 2 | FLOWS | 8 | Parallel DAG, research fan-out, pipeline, recall-driven, docgen, repo scan, stage deps, pool verification | FlowGenerator #170 |
| 3 | AGENTS | 10 | Lifecycle+memory, knowledge, timer+stuck, events, exportable, requires, transitions, spawn, persistent, multi-agent | AgentGenerator #171 |
| 4 | SYSTEMS | 11 | Composition, pool, warden, contract, states, events, types, endpoints, boundary, complete system | SystemAssembler #172 |
| 5 | ERRORS | 12 | Pure violations (8 types: reason/classify/escalate/exec/recall/learn/spawn/find), boundary, uncertain, states, warden | RepairAgent #173 |
| 6 | TESTING | 8 | Parser ok/error, checker ok/error, runtime mock/trace, error format, inventory | TestGenerator #174 |

### Design Principles

- **Embedded heredocs** — curated patterns are stable, reviewable, independent of source file changes
- **Keyword-rich format** — optimized for TF-IDF recall (tokenizer splits on non-alphanumeric, lowercases, filters <2 chars)
- **Cross-referenced** — each fact mentions related categories/patterns
- **Ascending complexity** — TASKS (atomic) → FLOWS (composition) → AGENTS (stateful) → SYSTEMS (orchestration)
- **Compounding knowledge** — seed curriculum bootstraps; continuous learning channels grow knowledge over time

### Verification

```bash
# Dry run
bash scripts/pretrain-toolkit.sh --dry-run

# Full ingestion
bash scripts/pretrain-toolkit.sh

# Verify recall confidence > 0.6
bash scripts/pretrain-toolkit.sh --verify-only
```

### Next Steps

- Run pretrain-toolkit.sh against a built forge-sensei binary
- Verify all 6 recall queries return confidence > 0.6
- Issue #167 will build the toolkit agent knowledge transfer infrastructure on top of this curriculum
